packages/core/react/src/WalletProvider.tsx

```diff
@@ -135,20 +135,20 @@ export const WalletProvider: FC<WalletProviderProps> = ({
             if (adapter) await adapter.disconnect();
             setName(walletName);
         },
-        [name, adapter, setName]
+        [name]
```

Comment:
Missing `adapter` as a dependency now 😅 Otherwise looks good to merge! 

And, if you do want to add a lint rule for this, that sounds good to me.

Reply:
![dammit-damn-it](https://user-images.githubusercontent.com/13243/147018037-b7f6dccf-f1a7-4fca-a6ef-de4f826a2dc9.gif)